internal/dkg/actions_signing.go

```diff
@@ -63,20 +69,90 @@ func (d *Process) verifyMessage(packet *drand.GossipPacket, metadata *drand.Goss
 	}
 
 	// we need to copy here or the GC/compiler does something weird
-	sig := make([]byte, len(metadata.Signature))
-	copy(sig, metadata.Signature)
-
-	msg := messageForProto(proposal, packet, metadata.BeaconID)
+	sig := make([]byte, len(packet.GetMetadata().GetSignature()))
+	copy(sig, packet.GetMetadata().GetSignature())
 
-	return kp.Scheme().AuthScheme.Verify(pubPoint, msg, sig)
+	msg := messageForSigning(packet, proposal)
+	err = kp.Scheme().AuthScheme.Verify(pubPoint, msg, sig)
+	if err != nil {
+		return fmt.Errorf("error verifying '%s' packet with msg '%s': %w ", packetName(packet), hex.EncodeToString(msg), err)
+	}
+	return nil
 }
 
-func messageForProto(proposal *drand.ProposalTerms, packet *drand.GossipPacket, beaconID string) []byte {
-	// we remove the metadata for verification of the packet, as the signer hasn't created the metadata
-	// upon signing
-	packetWithoutMetadata := proto.Clone(packet).(*drand.GossipPacket)
-	packetWithoutMetadata.Metadata = nil
-	return []byte(proposal.String() + packetWithoutMetadata.String() + beaconID)
+func messageForSigning(packet *drand.GossipPacket, proposal *drand.ProposalTerms) []byte {
+	// bytes.Buffer Writes are always returning nil errors, no need to check them
+	var ret bytes.Buffer
+	// we validate the packet type
+	ret.WriteString("sender:" + packet.GetMetadata().GetAddress())
+	ret.WriteString("beaconID:" + packet.GetMetadata().GetBeaconID())
+
+	switch t := packet.Packet.(type) {
+	case *drand.GossipPacket_Proposal:
+		ret.WriteString("Proposal:")
```

Comment:
perhaps i'm missing something but this seems to be duplicated below

Reply:
Below it's using the proposal itself, not the packet